### ISSUE TITLE
Channel display name not replacing #-style name

### ISSUE THREAD

**Issue by Valinwolf** [matrix-construct/construct#110]
![screenshot](https://drive.w0lfr.net/index.php/s/aetFiE8xwBWizwH/preview)

What should say "`Construct Test`" says "`#construct:flobob.ovh`"

**Comment by jevolk** [comment]
If you don't mind, can you open up devtools in Riot and check the room state to see if it contains an `m.room.name` and `m.room.canonical_alias`? Room name calculations are actually part of the spec, and complicated enough to still remain a mystery to me.

**Comment by Valinwolf** [comment]
> If you don't mind, can you open up devtools in Riot and check the room state to see if it contains an `m.room.name` and `m.room.canonical_alias`? Room name calculations are actually part of the spec, and complicated enough to still remain a mystery to me.
> 
> Also try to Clear Cache and reload and see if the name is recalculated

For the kicks, shortly after making this post I used the room settings to edit the name. I was surprised to find it _actually_ let me set the name. As such it does have `m.room.name`, but not `m.room.canonical_alias`. It also has not loaded a channel avatar for it, as you can see in the screenshot, which is probably related to the same underlying issue.

As for clear cache and reload, I've done it about (give or take a half dozen) 9 times since I joined the channel with no change in the channel. Anytime I find an issue I hit that button, about half the time it actually fixes the issue lol.

**Comment by jevolk** [comment]
> I was surprised to find it actually let me set the name.

This is not rejected with an error while construct is in development, my apologies for it being unexpected.

> As such it does have m.room.name, but not m.room.canonical_alias.

Yes, I'm really not sure why Riot sees the `m.room.name` but doesn't apply it. There is likely more Construct can do to feed Riot exactly what it expects from Synapse and when and how it expects it, but I don't know at this point.

> It also has not loaded a channel avatar for it, as you can see in the screenshot, which is probably related to the same underlying issue.

But it does have an `m.room.avatar` event present in the room state?

**Comment by jevolk** [comment]
I wrote my last message without having noticed that your address bar says `riot.im`. Are you using the matrix.org server in that screenshot?

**Comment by Valinwolf** [comment]
No, I'm using my Construct server in that screenshot so, yes, it is a bug using Riot on a Construct server.

**Comment by Valinwolf** [comment]
> > As such it does have m.room.name, but not m.room.canonical_alias.
> 
> Yes, I'm really not sure why Riot sees the `m.room.name` but doesn't apply it. There is likely more Construct can do to feed Riot exactly what it expects from Synapse and when and how it expects it, but I don't know at this point.

I think there's a misunderstanding... there is only `m.room.name` because _I set it_. You can see in the event details, that it was actually me that set it:
```json
{
  "content": {
    "name": "Construct Test"
  },
  "depth": 40753,
  "event_id": "$TIhmVTrf_SkWYhvT5w-9RVS1TrLUhEmr8r1BsdjXRU8:w0lfr.net",
  "origin_server_ts": 1564666308896,
  "room_id": "!AAAANTUiY1fBZ230:zemos.net",
  "sender": "@solinwolf:w0lfr.net",
  "state_key": "",
  "type": "m.room.name",
  "unsigned": {
    "age": 90
  }
}
```
